packages/teleport/src/Databases/AddDatabase/AddDatabase.test.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2021 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { fireEvent, render, screen } from 'design/utils/testing';
+import Manually from './Manually';
+
+describe('correct database add command generated with given input', () => {
+  test.each`
+    input                     | output
+    ${'self-hosted mysql'}    | ${'teleport start --roles=db --token=[generated-join-token] --auth-server=localhost:443 --db-name=[db-name] --db-protocol=mysql --db-uri=[uri]'}
+    ${'rds mysql'}            | ${'teleport start --roles=db --token=[generated-join-token] --auth-server=localhost:443 --db-name=[db-name] --db-protocol=mysql --db-uri=[uri] --db-aws-region=[region]'}
+    ${'self-hosted postgres'} | ${'teleport start --roles=db --token=[generated-join-token] --auth-server=localhost:443 --db-name=[db-name] --db-protocol=postgres --db-uri=[uri]'}
+    ${'rds postgres'}         | ${'teleport start --roles=db --token=[generated-join-token] --auth-server=localhost:443 --db-name=[db-name] --db-protocol=postgres --db-uri=[uri] --db-aws-region=[region]'}
+    ${'cloud sql postgres'}   | ${'teleport start --roles=db --token=[generated-join-token] --auth-server=localhost:443 --db-name=[db-name] --db-protocol=postgres --db-uri=[uri] --db-ca-cert=[instance-ca-filepath] --db-gcp-project-id=[project-id] --db-gcp-instance-id=[instance-id]'}
+    ${'redshift'}             | ${'teleport start --roles=db --token=[generated-join-token] --auth-server=localhost:443 --db-name=[db-name] --db-protocol=postgres --db-uri=[uri] --db-aws-region=[region] --db-aws-redshift-cluster-id=[cluster-id]'}
+  `(
+    'should generate correct command for input: $input',
+    ({ input, output }) => {
+      render(<Manually user="yassine" version="5.0.0" onClose={() => null} />);
+
+      const startCmd = screen.getByText(/teleport start/);
+      const dropDownInputEl = document.querySelector('input');
+
+      fireEvent.change(dropDownInputEl, { target: { value: input } });
+      fireEvent.focus(dropDownInputEl);
+      fireEvent.keyDown(dropDownInputEl, { key: 'Enter', keyCode: 13 });
+
+      expect(startCmd.innerHTML).toBe(output);
```

Comment:
I think a better way is to just use `expect(screen.getByText(output)).toBeTruthy();`. I tested it and it works perfectly. If the element isn't found, it will give us `null` which is falsy, and the test will fail. One thing to note is the linter throws a warning telling us to avoid using `toBeTruthy`, though in our case I think it's fine. Let me know what you think :slightly_smiling_face:

Reply:
are you sure it returns a null? https://testing-library.com/docs/queries/about/#types-of-queries

also there is `not.tobenull` i believe, if a linter complains there is usually another (better) way

Reply:
So `getBy` throws an error instead of null, however the test still passed since it will only pass if it's a truthy value, such as an element node. There's a better way to do it though. We can change it to `queryByText` instead and that will return `null` if the element isn't found, and then we can use `not.toBeNull()` to check that the element has been found :slightly_smiling_face: 

`expect(screen.queryByText(output)).not.toBeNull();`

Reply:
`queryByText` returns null if not found, but can return more than one. This means we need to assert that only one exists (b/c that's what we are expecting) 

```
const el = screen.queryByText(output)
expect(el).not.toBeNull()
expect(el.toHaveLength(1)
```
There are use cases for queryByText, but the most useful one is when we want to test that something does NOT exist, b/c its simple to test for a null: https://gravitational.slab.com/posts/testing-guideline-proposal-wip-yuc9uvwa#use-query-by-to-assert-that-something-is-not-rendered

Reply:
Btw `expect(screen.getByText(output)).not.toBeNull();` one liner is okay

Reply:
Since I'm also going to check that only one exists, I'll put in a variable `el` instead of the one liner

Reply:
I'll make these changes as part of the next PR